header-rewrite-filter/header_processor.h

```diff
@@ -10,7 +12,13 @@ namespace Extensions {
 namespace HttpFilters {
 namespace HeaderRewriteFilter {
 
-class HeaderProcessor {
+class Processor {
+public:
+  virtual ~Processor() {}
+  virtual absl::Status parseOperation(std::vector<absl::string_view>& operation_expression) = 0;
```

Comment:
nit: use `PURE` instead

Reply:
Created a dummy implementation for each pure virtual method.